pkg/mimirtool/analyze/grafana.go

```diff
@@ -109,6 +109,25 @@ func metricsFromTemplating(templating minisdk.Templating, metrics map[string]str
 			continue
 		}
 
+		if templateVar.Datasource.LegacyName != "" {
+			var isPrometheusDS bool
+			for _, templateVarTemp := range templating.List {
+				if templateVarTemp.Name == strings.Replace(templateVar.Datasource.LegacyName, "$", "", 1) && templateVarTemp.Type == "datasource" {
+					if templateVarTemp.Type == "prometheus" {
```

Comment:
I wonder is this covers all cases? (Mimir, Cortex, Thanos...)

Reply:
the type refers to the plugin type and not the prometheusType
Example:
```
grafana=> select distinct(type) from data_source order by 1;
               type
-----------------------------------
 alertmanager
 alexanderzobnin-zabbix-datasource
 elasticsearch
 grafadruid-druid-datasource
 grafana-clickhouse-datasource
 grafana-influxdb-flux-datasource
 grafana-opensearch-datasource
 grafana-sentry-datasource
 grafana-simple-json-datasource
 graphite
 influxdb
 jaeger
 loki
 marcusolsson-json-datasource
 mssql
 mysql
 opentsdb
 postgres
 prometheus
 tempo
 testdata
 vertamedia-clickhouse-datasource
```

The prometheusType is stored in the jsonData, example

```
grafana=> select type, json_data from data_source where type = 'prometheus' and org_id = 1;
    type    |                                                           json_data
------------+-------------------------------------------------------------------------------------------------------------------------------
 prometheus | {"alertmanagerUid":null,"exemplarTraceIdDestinations":null,"httpMethod":"POST","prometheusType":"Mimir","timeInterval":"15s"}
```

Would you be able to confirm that by testing against [grafana API](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/)?

Reply:
Thanks for jumping in @wilfriedroset, I haven't found an API resource that returns the possible types, but looking at docs only gives more confirmation that any Prometheus Query API backend fits into this `prometheus` datasource type, which is what we need. So I am confortable with this, if you are.

Reply:
Just checking the `prometheus` type should be fine here. Mimir, Cortex, Thanos & others are all queried using the `prometheus` data source.